internal/core/items/text.rs

```diff
@@ -595,9 +596,15 @@ impl Item for TextInput {
                 #[cfg(not(target_os = "android"))]
                 self.ensure_focus_and_ime(window_adapter, self_rc);
             }
-            MouseEvent::Released { button: PointerEventButton::Left, .. } => {
+            MouseEvent::Released { position, button: PointerEventButton::Left, .. } => {
+                let released_offset = self.byte_offset_for_position(position, window_adapter) as i32;
                 self.as_ref().pressed.set(0);
                 self.copy_clipboard(window_adapter, Clipboard::SelectionClipboard);
+                
+                if self.anchor_position_byte_offset() != released_offset {
```

Comment:
I think this is kind of not the right condition, as it shows in the test. Maybe it should be if there is a selection at all?